card/mod/history/lib/card/action/admin.rb

```diff
@@ -8,17 +8,6 @@ def delete_cardless
         joins(left_join).where("cards.id IS NULL").delete_all
       end
 
-      # permanently delete all {Action actions} not associated with a
-      # {Change change}
-      def delete_changeless
-        joins(
-          "LEFT JOIN card_changes "\
-          "ON card_changes.card_action_id = card_actions.id"
-        ).where(
-          "card_changes.id IS NULL"
-        ).delete_all
-      end
-
```

Comment:
A changeless action used to be a data problem. It's now the default case, and we pretty much never want to delete them.  I don't think it's wise even to offer the api to do so.